docs/guides/05-building/cross-platform.md

```diff
@@ -4,4 +4,145 @@ sidebar_position: 2
 
 # Cross-Platform Compilation
 
-How to use GH Action for Building: a glance at Tauri Action.
+Tauri relies heavily on native libraries and toolchains, so meaningful cross-compilation is not possible at the current moment. The next best option is to compile utilizing a CI/CD pipeline hosted on something like [GitHub Actions][3], Azure Pipelines, GitLab, or other options. The pipeline can run the compilation for each platform simultaneously making the compilation and release process much easier. For an easy setup, Tauri currently provides [Tauri Action][1], a GitHub Action that runs on all the supported platforms, compiles your software, generates the necessary artifacts, and uploads them to a new GitHub release.
+
+## How to use GH Action for Building: a glance at Tauri Action
+
+### Introduction
+
+Tauri Action leverages GitHub Actions to simultaneously build your web application as a Tauri native binary for macOS, Linux, and Windows, and automates putting it up on GitHub releases.
+
+This GitHub Action may also be used as a testing pipeline for your Tauri app, guaranteeing compilation runs fine on all platforms for each pull request sent, even if you don't wish to create a new release.
+
+### How to get started
+
+To set up Tauri Action you must first set up a GitHub repository, you can use this action on a repo that doesn't have Tauri configured since we automatically initialize Tauri before building, and configure it to use your Web artifacts.
+
+Go to the [Tauri Action GitHub page][1]. Then head to GitHub Action on your GitHub project and choose "New workflow", then choose "Set up a workflow yourself". Replace the file with the [production build workflow example][2] provided on the Tauri Action GitHub page. Alternatively, you may set up the workflow based on the example below on this page: [example workflow][6]
+
+### Configuration
+
+- You can configure Tauri with the `configPath`, `distPath` and `iconPath` options.
+- You can run custom Tauri CLI scripts with the `tauriScript` option. So instead of running `yarn tauri build` or `npx tauri build`, we'll execute `${tauriScript}`.
+  - Useful when you need custom build functionality when creating Tauri apps e.g. a `desktop:build` script.
+- When your app isn't on the root of the repo, use the `projectPath` input.
+
+- You may modify the workflow name, change the triggers, and add more steps such as npm run lint or npm run test. The important part is that you keep the:
+
+```yaml
+- uses: tauri-apps/tauri-action@v0
+```
+
+line at the end of the workflow, since this runs the build script and releases the artifacts.
```

Comment:
```suggestion
- You may modify the workflow name, change the triggers, and add more steps such as npm run lint or npm run test. The important part is that you keep the below line at the end of the workflow, since this runs the build script and releases the artifacts:

```yaml
- uses: tauri-apps/tauri-action@v0
```
```

I'm not 100% sure if this comment comes through correctly because of the double triple-ticks.